eureka2-core/src/main/java/com/netflix/eureka2/registry/NotifyingInstanceInfoHolder.java

```diff
@@ -227,19 +230,35 @@ public void call(Throwable throwable) {
         }
 
         logger.debug("REMOVE result: {}, source: {}", result, source);
-        return Observable.just(result);
+        return result;
     }
 
     @Override
     public String toString() {
         return "NotifyingInstanceInfoHolder{" +
                 "notificationSubject=" + notificationSubject +
-                ", dataMap=" + dataMap +
+                ", dataStore=" + dataStore +
                 ", id='" + id + '\'' +
                 ", snapshot=" + snapshot +
                 "} " + super.toString();
     }
 
+    /**
+     * @return true if the two sources have the same origin and name. Don't use a matcher from source
+     *         to avoid the object creation.
+     */
+    private boolean matches(Source one, Source two) {
```

Comment:
This is a somewhat misleading match specific to the holder, which is why it is here as a private method. The Source class already have matcher methods that does similar with a more sensible API.